src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.ServiceConnection.ps1

```diff
@@ -1056,7 +1060,78 @@ class ServiceConnection: ADOSVTBase
+                }
 
+                 $approvers = $approvalControl.settings.approvers | Select @{n='Approver name';e={$_.displayName}},@{n='Approver id';e = {$_.uniqueName}}
+                 $formattedApproversTable = ($approvers| FT -AutoSize | Out-String -width 512)
+                 
+                 if($approvalControl.Count -gt 0)
+                 {
+                    # match all the identities added on service connection with defined restricted list
+                     $restrictedGroups = $approvalControl.settings.approvers | Where-Object { $restrictedBroaderGroupsForSerConn -contains $_.displayName.split('\')[-1] } | select displayName
+                     
+                    # fail the control if restricted group found on service connection
+                    if($restrictedGroups)
+                    {
+                        $controlResult.AddMessage("Count of broader groups that have been added as approvers to service connection: ", @($restrictedGroups).Count)
+                        $controlResult.AddMessage([VerificationResult]::Failed,"Do not grant broader groups that have been added as approvers to service connections. Granting elevated permissions to these groups can risk exposure of service connections to unwarranted individuals.");
+                        $controlResult.AddMessage("Broader groups that have been added as approvers to service connection.",$restrictedGroups)
+                        $controlResult.SetStateData("Broader groups that have been added as approvers to service connection",$restrictedGroups)
+                        $controlResult.AdditionalInfo += "Count of broader groups that have been added as approvers to service connection: " + @($restrictedGroups).Count;
+                    }
+                    else{
+                        $controlResult.AddMessage([VerificationResult]::Passed,"No broader groups that have been added as approvers to service connection.");
+                        $controlResult.AddMessage("`nList of approvers : `n$formattedApproversTable");
+                        $controlResult.AdditionalInfo += "List of approvers on service connection  $($approvers).";
+                    }
+                }
+                else {
+                    $controlResult.AddMessage([VerificationResult]::Passed,"No broader groups that have been added as approvers to service connection.");
+                    $controlResult.AddMessage("`nList of approvers : `n$formattedApproversTable");
+                    $controlResult.AdditionalInfo += "List of approvers on service connection  $($approvers).";
+                }   
+            }  
+            $displayObj = $restrictedBroaderGroupsForSerConn | Select-Object @{Name = "Broader Group"; Expression = {$_}}
+            $controlResult.AddMessage("`nNote:`nThe following groups are considered 'broad' which should not have excessive permissions: `n$($displayObj | FT | out-string -width 512)`n");                  
```

Comment:
The following groups are considered 'broader' groups which should not be added as approvers:
Similar in var grp

Reply:
Message updated